src/ble/CHIPBleServiceData.h

```diff
@@ -64,17 +67,34 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint8_t GetAdvertisementVersion() const
+    {
+        uint8_t advertisementVersion = static_cast<uint8_t>(DeviceDiscriminatorAndAdvVersion[1] & ~kAdvertisementVersionMask);
```

Comment:
Shouldn't we shift first, then mask (i.e. undo exactly what we did when we initially set it)?  If we do it this way, we are relying on `kAdvertisementVersionMask` and `~kAdvertisementVersionMask` being the same number of bits, no?

Reply:
Yes updated